### PR TITLE
Configuration block

### PR DESCRIPTION
**Pull request by TadeasKriz**

### CODE REVIEW

File: src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftExtension.groovy [hunk 1]
```diff
@@ -109,30 +115,38 @@ class SpaceliftExtension {
...
-     def propertyMissing(String name) {
-
+    def propertyMissing(String name) {
         // try all containers to find resolution in particular order
         //order here defines order of reference in case reference to the same object is found, laters are ignored
         def object, objectType
-        for(def container : ([installations, tests, tools, profiles])) {
+
+        def containers = [configuration, installations, tests, tools, profiles]
+        if (project.hasProperty("selectedProfile")) {
```
Comment (kpiwko) [inline src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftExtension.groovy]: Can you please add this marker there? https://github.com/arquillian/arquillian-spacelift-gradle-plugin/issues/50

File: src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy [hunk 2]
```diff
@@ -142,6 +148,7 @@ class SpaceliftPlugin implements Plugin<Project> {
             task << {
                 println "Spacelift Workspace:          ${project.spacelift.workspace}"
                 println "Spacelift Installations Cache Dir:  ${project.spacelift.installationsDir}"
+                println "Spacelift configuration: \n${project.spacelift.configuration.toString()}"
```
Comment (kpiwko) [inline src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy]: Will this print out configuration of activated profile?

Comment (TadeasKriz) [inline src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy]: No, only the main configuration block. Each profile block then has its own configuration block. The description is not aware of selected profile afaik.

File: src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy [hunk 3]
```diff
@@ -306,4 +313,42 @@ class SpaceliftPlugin implements Plugin<Project> {
             }
         }
     }
+
+    private void loadConfiguration(Project project, Object logger) {
+        List<ConfigurationItem<?>> configurationItems = new ArrayList<>()
+
+        configurationItems.addAll(project.selectedProfile.configuration)
+
+        project.spacelift.configuration.each { ConfigurationItem<?> item ->
```
Comment (kpiwko) [inline src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy]: Suggestion: add a comment to this block

File: src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy [hunk 4]
```diff
@@ -306,4 +313,42 @@ class SpaceliftPlugin implements Plugin<Project> {
...
+                profileItem.name.equals(item.name)
+            }
+            if(profileItem == null) {
+                configurationItems.add(item);
+            } else if(!item.type.resolve().isAssignableFrom(profileItem.type.resolve())) {
+                throw IllegalConfigurationException.incompatibleOverride(item, profileItem)
+            }
+        }
+
+        configurationItems.sort(false) { ConfigurationItem<?> a, ConfigurationItem<?> b ->
```
Comment (kpiwko) [inline src/main/groovy/org/arquillian/spacelift/gradle/SpaceliftPlugin.groovy]: Suggestion: add a comment to this block

File: src/main/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConverters.groovy [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        String toString(Class<?> value) {
+            return value.getName()
+        }
+    }
+
+    static class FileConfigurationItemConverter implements ConfigurationItemConverter<File> {
+
+        @Override
+        File fromString(String value) throws ConversionException {
+            return new File(value)
```
Comment (kpiwko) [inline src/main/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConverters.groovy]: what happens if value is not absolute?

I'd expect this is rather new File(spacelift.workspace,  value)

File: src/test/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConvertersTest.groovy [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+/**
+ * Created by tkriz on 25/03/15.
+ */
+class BuiltinConfigurationItemConvertersTest {
+
+    @Test
+    void "Test boolean converter"() {
+        def converter = BuiltinConfigurationItemConverters.getConverter(Boolean)
+
+        assertThat(converter.toString(Boolean.FALSE), is('false'))
```
Comment (kpiwko) [inline src/test/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConvertersTest.groovy]: Please remove parentheses, we are not using them in groovy tests, e.g.

```
assertThat converter.toString(Boolean.FALSE), is('false')
```

File: src/main/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConverters.groovy (hunk 5 above)
Comment (TadeasKriz) [inline src/main/groovy/org/arquillian/spacelift/gradle/configuration/BuiltinConfigurationItemConverters.groovy]: The converter doesn't have access to spacelift. Should we pass it in as well?

**Comment by kpiwko** [comment]
Squashed and merged in https://github.com/arquillian/arquillian-spacelift-gradle-plugin/commit/26eb1896a309e0f5e5eb1f932bebdc89e4a7cbaa.

Thanks!

**Pull request closed without merging**